### PR TITLE
Fix: Concatinate multiple artist names w/ unittest

### PR DESCRIPTION
**Pull request by krak3n**
This PR fixes the API movement of artist data from the album object to directly on the track object. Also uses list comprehension to contacting multiple artists names together.

### CODE REVIEW

File: fmslack/cli.py [hunk 1]
```diff
@@ -114,9 +115,11 @@ def slack_post(slack_webhook_url, name, artist, album, image):
 
     logger.debug('Posting to Slack webhook {0}'.format(slack_webhook_url))
 
+    artists = ' & '.join([artist['name'] for artist in artists])
```
Comment (krak3n) [inline fmslack/cli.py:118]: List comprehensions (along with dict comprehensions) are one of the awesome features of python and very powerful when combined with other batteries included features, like string joining as we are doing here.

File: tests/test_slack.py [hunk 2]
```diff
@@ -49,32 +50,52 @@ def test_tracks_request_error(self):
 
 class TestSlackPost(BaseTestCase):
 
+    track = {
+        'name': 'track name',
+        'artists': [{'name': 'artist 1'}, {'name': 'artist 2'}],
+        'album': 'album name',
+        'image': 'http://albumimage.url'
+    }
```
Comment (krak3n) [inline tests/test_slack.py:58]: Moving this up here saves on repetition.

File: tests/test_slack.py [hunk 3]
```diff
@@ -49,32 +50,52 @@ def test_tracks_request_error(self):
...
     def test_slack_post(self):
+        post = mock.MagicMock(status_code=200)
+        self.requests.post.return_value = post
 
+        slack_post(
+            'http://slack.com',
+            self.track['name'],
+            self.track['artists'],
+            self.track['album'],
+            self.track['image'])
```
Comment (krak3n) [inline tests/test_slack.py:69]: Just looks neater and is easier to read at a glance.

**Comment by jamesjwarren** [comment]
Nice, thanks :+1:

**Pull request merged**